prometheus/registry.go

```diff
@@ -505,22 +470,30 @@ func (r *registry) writePB(encoder expfmt.Encoder) error {
 		sort.Sort(metricSorter(mf.Metric))
 	}
 
-	// Write out MetricFamilies sorted by their name.
+	// Write out MetricFamilies sorted by their name, skipping those without
+	// metrics.
```

Comment:
I see. Right now, that would create an error in encoding (and possibly decoding - not sure how the text parser deals with that). This needs a more basic approach. First the expfmr package has to be changed, and then it has to propagate into all Go binaries (server, PGW) before we can allow it in the client.